### DISCUSSION TITLE
NetStream bufferTime

### DISCUSSION THREAD

**Discussion by koreaflash** [airsdk/Adobe-Runtime-Support#1885] [General]
Streams a video with http:// using a class extending NetStream and Socket.
Play with the initial bufferTime set to 2 seconds.
But after NetStream.Buffer.Empty state, bufferTime is automatically changed from 2 seconds to 5 seconds. Also, the next time NetStream.Buffer.Empty occurs, the bufferTime will automatically change from 5 seconds to 10 seconds.
Is there any way to freeze the bufferTime?

If you set bufferTime again after starting playback, it will not be applied. Is there any way to apply it?

**Comment by 2jfw** [comment]
Is the class you are referring to Greensocks `VideoLoader`?
If so, there is a property called `autoAdjustBuffer` which is `true` by default:

> If the buffer becomes empty during playback and <code>autoAdjustBuffer</code> is <code>true</code> (the default), it will automatically attempt to adjust the NetStream's <code>bufferTime</code> based on the rate at which the video has been loading, estimating what it needs to be in order to play the rest of the video without emptying the buffer again. This can prevent the annoying problem of video playback start/stopping/starting/stopping on a system tht doesn't have enough bandwidth to adequately buffer the video. You may also set the <code>bufferTime</code> in the constructor's <code>vars</code> parameter to set the initial value.</li>

**Reply by koreaflash** [reply]
Play video using flash.net.Socket, flash.net.NetStream, flash.media.Video class.
